source/common/network/io_socket_handle_impl.cc

```diff
@@ -517,11 +524,21 @@ Address::InstanceConstSharedPtr IoSocketHandleImpl::peerAddress() {
   return Address::addressFromSockAddr(ss, ss_len);
 }
 
-Event::FileEventPtr IoSocketHandleImpl::createFileEvent(Event::Dispatcher& dispatcher,
-                                                        Event::FileReadyCb cb,
-                                                        Event::FileTriggerType trigger,
-                                                        uint32_t events) {
-  return dispatcher.createFileEvent(fd_, cb, trigger, events);
+void IoSocketHandleImpl::createFileEvent(Event::Dispatcher& dispatcher, Event::FileReadyCb cb,
+                                         Event::FileTriggerType trigger, uint32_t events) {
+  RELEASE_ASSERT(file_event_ == nullptr, "Attempting to initialize two `file_event_` for the same "
```

Comment:
Is there a case that two worker threads can race to create the `file_event_`? In that case this would be UB.

Reply:
added an ASSERT in a38da8e68315fb2524f0654006b30ef4ca6bb42c